src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h

```diff
@@ -0,0 +1,745 @@
+
+    /** Index range of the joint torque */
+    iDynTree::IndexRange m_jointTorqueIndex{iDynTree::IndexRange::InvalidRange()};
+
+    iDynTree::MatrixDynSize m_massMatrix; /**< Floating-base mass matrix  */
+    iDynTree::FreeFloatingGeneralizedTorques m_generalizedBiasForces; /**< Coriolis and
+                                                                         Gravitational term  */
+    iDynTree::MatrixDynSize m_jacobianMatrix; /**< Jacobian Matrix  */
+
+    std::vector<Frame> m_framesInContact; /**< Vectors containing the frames in contact with the
+                                             environment */
+
+    iDynTree::MatrixDynSize m_reflectedInertia; /**< Reflected inertia matrix  */
+    bool m_useReflectedInertia; /**< If true the reflected inertia will be used  */
+
+public:
+    /**
+     * Constructor. If you call this constructor the motor reflected inertia is disabled
+     * @param kinDyn an iDynTree kinDyn computation object
+     * @param handler the variable handler object
+     * @param framesInContact vector containing the frames in contact. Each element of the vector is
+     * a pair containing the name of the frame in the variableHandler and in the model.
+     * @throw std::runtime_error if the frame is not defined
+     */
+    SystemDynamicsElement(iDynTree::KinDynComputations& kinDyn,
+                          const VariableHandler& handler,
+                          const std::vector<std::pair<std::string, std::string>>& framesInContact);
+
+    /**
+     * Constructor. If you call this constructor the motor reflected inertia is enable
+     * @param kinDyn an iDynTree kinDyn computation object
+     * @param handler the variable handler object
+     * @param framesInContact vector containing the frames in contact. Each element of the vector is
+     * a pair containing the name of the frame in the variableHandler and in the model.
+     * @param gamma diagonal matrix containing the gear box ratios;
+     * @param motorsInertia vector containing the inertia of the motors
+     * @param harmonicDriveInertia vector containing the inertia of the harmonic drive
+     * @param r torso parameter http://wiki.icub.org/wiki/ICub_coupled_joints
+     * @param R torso parameter http://wiki.icub.org/wiki/ICub_coupled_joints
+     * @param t shoulder parameter http://wiki.icub.org/wiki/ICub_coupled_joints
+     * @throw std::runtime_error if the frame is not defined
+     * @note this is iCub specific. In the next version it will be removed
+     */
+    [[deprecated(
+        "This function is iCub specific. In the next implementation will be removed. If "
+        "you want to include regularization element for the mass matrix please call "
+        "SystemDynamicsElement(iDynTree::KinDynComputations&, const VariableHandler&, "
+        "const std::vector<std::pair<std::string, std::string>>&, const "
+        "iDyntree::MatrixDynsize&)")]]
+    SystemDynamicsElement(iDynTree::KinDynComputations& kinDyn,
```

Comment:
How comes that you have already deprecated methods? :sweat_smile:

Reply:
As discussed F2F I remove this method

Reply:
Removed in 1e4e9dcbe2da2e3392201a0aa53f3a80968d84b1